lib/product_list/cubits/product_list_cubit.dart

```diff
@@ -30,4 +31,34 @@ class ProductListCubit extends Cubit<ProductListState> {
       ));
     }
   }
+
+  Future<void> sortProducts(
+      {String? categoryCode, required Sort sortBy}) async {
+    try {
+      emit(ProductListState(
+        status: Status.pending,
+        searchResults: state.searchResults,
+        products: state.products,
+      ));
+
+      final searchResults =
+          await GetIt.instance.get<ProductListApi>().fetchProducts(
+                categoryCode: state.searchResults!.currentQuery.lastLeaf,
+                sortBy: sortBy.code,
+              );
+      final products = searchResults.products;
```

Comment:
`products` has a special treatment, why not destructuring the other fields in the cubit state?